src/core/resolve.c

```diff
@@ -1833,6 +1847,23 @@ ip_addr_t *str2ip(str *st)
 	return ipb;
 }
 
+struct hostent *__resolvehost(char *name)
+{
+	if(dns_cache_init) {
+		return dns_resolvehost(name);
+	} else {
+		return _resolvehost(name);
+	}
+}
+
+struct hostent *__sip_resolvehost(str *name, unsigned short *port, char *proto)
+{
+	if(dns_cache_init) {
+		return dns_sip_resolvehost(name, port, proto);
+	} else {
```

Comment:
This `else` can be skipped:
```c
if (x) return y;
/*skipped else*/ return z;
```

Reply:
True, both way are valid though. I think it's clearer with the `else`, but if maintainers have a preference I am fine to change that.